docs: initial draft for restructuring the CMS quickstarts guide

rpkoller opened the pull request:
## The Issue
https://github.com/ddev/ddev/issues/5206

## How This PR Solves The Issue

## Manual Testing Instructions

## Automated Testing Overview

<!-- Please describe the tests introduced by this PR, or explain why no tests are needed. -->

## Related Issue Link(s)

## Release/Deployment Notes

<!-- Does this affect anything else or have ramifications for other code? Does anything have to be done on deployment? -->



<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/5207"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

mattstein commented:
I like this! Feels a lot more structured, maintainable, and easier to navigate. What level of feedback is most helpful with this right now?

rfay commented:
Changed to draft/WIP, let us know when you want a review.

rfay reviewed (commented):
Looking through https://ddev--5207.org.readthedocs.build/en/5207/users/quickstart/ it seems like this is going to work out fine. The only question is going to be whether you want to go forward before or after silverstripe goes in.

rfay commented:
This will be now slightly broken due to the new silverstripe pull.

rpkoller commented:
> I like this! Feels a lot more structured, maintainable, and easier to navigate. What level of feedback is most helpful with this right now?

Apologies for the late reply. The general idea for this PR was to untwine the current state and move changes to the quickstarts to one or two follow up issues. So in regards of feedback I would consider the following points helpful (but feel free to add anything else you notice): 

- https://github.com/ddev/ddev/pull/5207/files in `docs/content/users/quickstart.md` I’ve updated line 3 and also added an additional line 5. The introduction in line 3 only covered PHP but was missing Python. So i’ve tried to add that there. The newly introduced line 5 could maybe moved to one of the follow up issues. It is basically about providing the most basic steps that apply to every quickstarts guide. I got the inspiration for that bit from the Typo3 documentation. Basically making sure that the user is aware that DDEV is up to date and that the person is aware that a new empty directory has to be created. That way all the different `mkdir my-directory` and `cd my-directory` lines in the different quickstarts could be striked and quickstarts could become more concise and consistent in one of the followup issues (maybe move the change of line 5 to one of the followup issues even). That is the basic idea behind the changes for the introductory section. i think the entire paragraph could use an in-depth feedback. 

-  content wise the rest hasn’t been touched at all (i would suggest to do that in the followup issue). in regards of further feedback the questions are:

1) for `docs/content/users/usage/managing-projects.md`: does it work moving the whole configuration files part over to it’s dedicate page or is it necessary to make some adjustments to that page content wise and is `Managing Projects` an appropriate title that is clear to the user? 

2) for `docs/content/users/quickstart.md` is it ok to move Statamic and Backdrop out of Laravel and Drupal. Cuz it was sort of inconsistent since Statamic isn’t a product done by the Laravel folks, it is just a CMS based on Laravel. The other way around there would have been a need of having a Symfony quickstarts section otherwise. I consider the suggested list of quickstarts the most straight forward. All the main CMS types are directly available and reachable via the table of contents. 

4) do all the links still work properly, linking in between documentation pages. 


> This will be now slightly broken due to the new silverstripe pull.

i’ve noticed that when i’ve updated DDEV head today. one question how to approach that. do i have to a) sync the branch with the main branch and make adjustments? or b) simply copy the section for silverstripe from the main branch?

rfay commented:
> i’ve noticed that when i’ve updated DDEV head today. one question how to approach that. do i have to a) sync the branch with the main branch and make adjustments? or b) simply copy the section for silverstripe from the main branch?

it's always a bit of a pain, but in this case you can probably just make sure that you haven't lost anything upstream, and then copy the silverstripe stuff (and update it) into your branch. Mostly when rebasing use your content. But make sure you're not losing anything else.

rpkoller commented:
ohh i've noticed syncing the branch won't work 
![Screen Shot 2023-08-10 at 21 50 35](https://github.com/ddev/ddev/assets/1665422/22337a72-a3f4-4485-a17e-0ee57e7c251d)
i will then try to simply  copy the silverstripe changes into the branch and push the changes up.

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
 # CMS Quickstarts
 
-While the generic `php` project type is [ready to go](./project.md) with any CMS or framework, DDEV offers project types for more easily working with popular platforms and content management systems:
+With the generic `php` and `python` project types DDEV is [ready to go](./project.md) with any PHP and Python based content management system (CMS) or framework. In addition there are preconfigured project types for many popular platforms and CMSes. To learn more about how to manage projects in DDEV visit the [Managing Projects](../users/usage/managing-projects.md) page.
```
There’s an echo in “With the generic...” and “...with any PHP...” and all of this could be more concise:

```suggestion
DDEV is [ready to go](./project.md) with generic project types for PHP and Python frameworks, and more specific project types for [working with popular platforms and CMSes](../users/usage/managing-projects.md).
```

rpkoller replied:
i like the changes for the first sentence. but i wouldnt strike the second and just leave in the link to the managing projects page. for me the managing projects page has a different purpose than what the first sentence is about. therefore i would consider it confusing. the first sentence is about what is available on the quickstarts page while the striked second sentence is about what the manageing projects page is about.

rpkoller replied:
I would suggest to bring the second sentence back in: 

```DDEV is [ready to go](./project.md) with generic project types for PHP and Python frameworks, and more specific project types for working with popular platforms and CMSes. To learn more about how to manage projects in DDEV visit the [Managing Projects](../users/usage/managing-projects.md) page.```

rfay replied:
It's your PR, so write it the way you think it should be and Matt will get another chance :)

mattstein replied:
> I would suggest to bring the second sentence back in:
> (...)

This reads better to me, and I see your point! Only nitpick left is that it could end with “visit Managing Projects” since that’d clearly link to a page—meaning “the” and “page” don’t add anything and can be removed.

rpkoller replied:
pushed a commit with your suggested changes.

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
+With the generic `php` and `python` project types DDEV is [ready to go](./project.md) with any PHP and Python based content management system (CMS) or framework. In addition there are preconfigured project types for many popular platforms and CMSes. To learn more about how to manage projects in DDEV visit the [Managing Projects](../users/usage/managing-projects.md) page.
 
-=== "Craft CMS"
+Before proceeding, make sure your installation of DDEV is up-to-date. In a new and empty project folder, using your favorite shell, run the following commands:
```
I think _up-to-date_ would function more like an adjective, so I’d drop the hyphens in this case:

```suggestion
Before proceeding, make sure your installation of DDEV is up to date. In a new and empty project folder, using your favorite shell, run the following commands:
```

rpkoller replied:
that is totally fine with me. i was just searching on google which of the variants (with and without the dashes) was more common.

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
         If you previously set up DDEV in a Craft project using the generic `php` project type, update the `type:` setting in `.ddev/config.yaml` to `craftcms`, then run [`ddev restart`](../users/usage/commands.md#restart) apply the changes.
 
-    ### Running Craft in a Sub-directory
+### Running Craft in a Sub-directory
```
Un-hyphenate _sub-directory_ to be more consistent with DDEV and Craft’s docs and [common usage](https://www.merriam-webster.com/dictionary/subdirectory):

```suggestion
### Running Craft in a Subdirectory
```

rpkoller replied:
and one comment about the rest of the changes. i haven't touched any of the quickstarts text for this PR. wanted to tackle that in the next steps. but all the suggested changes are definitely improvements.

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
-    ### Running Craft in a Sub-directory
+### Running Craft in a Sub-directory
+
+In order for `ddev craft` to work when Craft is installed in a sub-directory, you will need to change the location of the `craft` executable by providing the `CRAFT_CMD_ROOT` environment variable to the web container. For example, if the installation lives in `my-craft-project/app`, you would run `ddev config --web-environment-add=CRAFT_CMD_ROOT=./app`. `CRAFT_CMD_ROOT` defaults to `./`, the proj ...
```
```suggestion
In order for `ddev craft` to work when Craft is installed in a subdirectory, you will need to change the location of the `craft` executable by providing the `CRAFT_CMD_ROOT` environment variable to the web container. For example, if the installation lives in `my-craft-project/app`, you would run `ddev config --web-environment-add=CRAFT_CMD_ROOT=./app`. `CRAFT_CMD_ROOT` defaults to `./`, the project root directory. Run `ddev restart` to apply the change.
```

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
+
+In order for `ddev craft` to work when Craft is installed in a sub-directory, you will need to change the location of the `craft` executable by providing the `CRAFT_CMD_ROOT` environment variable to the web container. For example, if the installation lives in `my-craft-project/app`, you would run `ddev config --web-environment-add=CRAFT_CMD_ROOT=./app`. `CRAFT_CMD_ROOT` defaults to `./`, the proj ...
+
+More information about customizing the environment and persisting configuration can be found in [Providing Custom Environment Variables to a Container](https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#providing-custom-environment-variables-to-a-container).
```
Could shorten and drop impersonal “can be found in”:

```suggestion
Read more about customizing the environment and persisting configuration in [Providing Custom Environment Variables to a Container](https://ddev.readthedocs.io/en/latest/users/extend/customization-extendibility/#providing-custom-environment-variables-to-a-container).
```

mattstein commented on docs/content/users/quickstart.md:
````diff
@@ -1,274 +1,230 @@
+ddev launch
+```
+
+* DDEV will install all everything in your `requirements.txt` or `pyproject.toml` into a `venv`. This takes a little while on first startup.
````
```suggestion
* DDEV will install everything in your `requirements.txt` or `pyproject.toml` into a `venv`. This takes a little while on first startup.
```

mattstein commented on docs/content/users/quickstart.md:
````diff
@@ -1,274 +1,230 @@
+    ```
 
-    ## Drupal
+    Drupal 7 doesn’t know how to redirect from the front page to `/install.php` if the database is not set up but the settings files *are* set up, so launching with `/install.php` gets you started with an installation. You can also `drush site-install`, then `ddev exec drush site-install --yes`.
````
Run commands:

```suggestion
    Drupal 7 doesn’t know how to redirect from the front page to `/install.php` if the database is not set up but the settings files *are* set up, so launching with `/install.php` gets you started with an installation. You can also run `drush site-install`, then `ddev exec drush site-install --yes`.
```

mattstein commented on docs/content/users/quickstart.md:
```diff
@@ -1,274 +1,230 @@
-=== "Laravel"
-
-    ## Laravel
+The Laravel project type can be used for [Lumen](https://lumen.laravel.com/) just as it can for Laravel. DDEV automatically updates or creates the `.env` file with the database information.
```
```suggestion
The Laravel project type can be used for [Lumen](https://lumen.laravel.com/) like it can for Laravel. DDEV automatically updates or creates the `.env` file with the database information.
```

mattstein commented on docs/content/users/quickstart.md:
````diff
@@ -312,422 +268,243 @@ While the generic `php` project type is [ready to go](./project.md) with any CMS
- ddev manages this file and may delete or overwrite the file unless this comment is removed.
- */
-```
+The Laravel project type can be used for [Statamic](https://statamic.com/) just as it can for Laravel. DDEV automatically updates or creates the `.env` file with the database information.
````
```suggestion
The Laravel project type can be used for [Statamic](https://statamic.com/) like it can for Laravel. DDEV automatically updates or creates the `.env` file with the database information.
```

rfay commented:
Congrats and thanks! Great to get this in! I hope it doesn't break too many links.
